### PR TITLE
feat(dev): alembic check

### PR DESCRIPTION
**Pull request by miketheman**
Implements wrapper for `check` command.

Refs: https://alembic.sqlalchemy.org/en/latest/autogenerate.html#running-alembic-check-to-test-for-new-upgrade-operations
Refs: https://alembic.sqlalchemy.org/en/latest/api/commands.html#alembic.command.check

Signed-off-by: Mike Fiedler <miketheman@gmail.com>

### CODE REVIEW

**Comment by miketheman** [comment]
Note: this check should **not be marked as required**, since there are cases where we want to allow for a code-side removal (which would generate a migration) and merge that PR, and ship the database migration after the code no longer relies on it.

**Comment by miketheman** [comment]
You can see a failing run here, where I introduce a code change that has no associated migration:
https://github.com/pypi/warehouse/actions/runs/10394102434/job/28783297165?pr=16483

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -93,3 +93,41 @@ jobs:
...
+        uses: falti/dotenv-action@v1
+        with:
+          path: dev/environment
+          export-variables: true
+          keys-case: upper
+      - name: Check Database
+        run: bin/db-check
+        env:
+          # override the hostname set in `dev/environment`
+          DATABASE_URL: 'postgresql+psycopg://postgres@postgres/warehouse'
```
Comment (di) [inline .github/workflows/ci.yml:133]:
```suggestion
      - name: Check for 'partial-migration' label
        id: check_label
        run: |
          if [[ "$(jq '.pull_request.labels[].name' <<<"$GITHUB_EVENT_PATH")" == *'"partial-migration"'* ]]; then
            echo "Label 'partial-migration' found, skipping database consistency check."
            exit 0
          fi
        continue-on-error: true

      - name: Check Database
        if: steps.check_label.outcome == 'success'
        run: bin/db-check
        env:
          DATABASE_URL: 'postgresql+psycopg://postgres@postgres/warehouse'
```

Comment (di) [inline .github/workflows/ci.yml:133]:
```suggestion
      - name: Check Database
        if: "! contains(github.event.pull_request.labels.*.name, 'partial-migration')"
        run: bin/db-check
        env:
          # override the hostname set in `dev/environment`
          DATABASE_URL: 'postgresql+psycopg://postgres@postgres/warehouse'
```
Might be possible to just do this instead, I'm not sure what the overall outcome of the job will be if the last check is skipped like this though.

**Review by di — APPROVED** [review]

**Pull request merged**